sanic/cli/app.py

```diff
@@ -125,9 +130,18 @@ def _get_app(self):
                 app_type_name = type(app).__name__
 
                 if not isinstance(app, Sanic):
+                    if hasattr(app, "__call__"):
+                        solution = f"sanic {self.args.module} --factory"
+                        raise ValueError(
+                            "Module is not a Sanic app, it is a"
+                            f"{app_type_name}\n"
+                            "  If your function returns a"
```

Comment:
Let's not presume that it is a function.

```
If this callable returns a
```

Reply:
done and dusted :)